### PR TITLE
Cv 127 frontend work

### PR DESCRIPTION
**Pull request by Pauljgraham**
Re-introduced date validation and validation messages on web page

### CODE REVIEW

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Models/AddDraftApprenticeshipViewModel.cs [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
 using System.ComponentModel.DataAnnotations;
+using FluentValidation.Attributes;
 using SFA.DAS.ProviderCommitments.Domain_Models.ApprenticeshipCourse;
 using SFA.DAS.ProviderCommitments.ModelBinding.Models;
 using SFA.DAS.ProviderCommitments.Models;
+using SFA.DAS.ProviderCommitments.Web.Requests;
 
 namespace SFA.DAS.ProviderCommitments.Web.Models
 {
+    [Validator(typeof(AddDraftApprenticeshipViewModelValidator))]
```
Comment (iangledhill) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Models/AddDraftApprenticeshipViewModel.cs]: We don't need this attribute - we've injected FV into the MVC pipeline.

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Requests/AddDraftApprenticeshipViewModelValidator.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    public class AddDraftApprenticeshipViewModelValidator : AbstractValidator<AddDraftApprenticeshipViewModel>
+    {
+        public AddDraftApprenticeshipViewModelValidator()
+        {
+            RuleFor(x => x.BirthDate).Must(y => y.IsValid).WithMessage("Date of birth is invalid").When(z => z.BirthDate.HasValue);
+            RuleFor(x => x.StartDate).Must(y => y.IsValid).WithMessage("Planned training start date is invalid").When(z => z.StartDate.HasValue);
+            RuleFor(x => x.FinishDate).Must(y => y.IsValid).WithMessage("Projected finish date is invalid").When(z => z.FinishDate.HasValue);
+        }
+    }
+}
```
Comment (PeterDavidWilliams) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Requests/AddDraftApprenticeshipViewModelValidator.cs]: This has no unit tests.

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Models/AddDraftApprenticeshipViewModel.cs (hunk 1 above)
Comment (Pauljgraham) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Models/AddDraftApprenticeshipViewModel.cs]: Good point

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Requests/AddDraftApprenticeshipViewModelValidator.cs (hunk 2 above)
Comment (Pauljgraham) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Requests/AddDraftApprenticeshipViewModelValidator.cs]: Added unit tests

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Models/AddDraftApprenticeshipViewModel.cs (hunk 1 above)
Comment (Pauljgraham) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Models/AddDraftApprenticeshipViewModel.cs]: addressed and removed duplicate IMediator registration from DefaultRegistry

**Pull request merged**